### PR TITLE
Price context migration to redux toolkit store

### PR DESCRIPTION
**Pull request by mateodaza**

### CODE REVIEW

**Review by MohammadPCh — CHANGES_REQUESTED** [review]
@mateodaza 
Thanks for your PR,
I have some suggestions that may help us:
1. remove serializableCheck middleware
2. store bignumbers as string in redux
3. use redux-thunk to create async functions and put them in extraReducer.
4. store 3 states in price slice (give price, eth price, and object of token prices)
5. add 3 extraReducer with thunk that calls an end-point and store data for each state in it.
6. use the price controller to get the real values or if we want to update them.

LMK if I can help :)

File: src/components/controller/price.ctrl.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+	} catch (e) {
+		console.error('Error fetching token price:', tokenAddress, e);
+		captureException(e, {
+			tags: {
+				section: 'fetUniswapSubgraphTokenPrice',
+			},
+		});
+		return Zero;
+	}
+};
```
[the remark this answers is not in the document]
Comment (mateodaza) [reply src/components/controller/price.ctrl.tsx:46]: This is fetching from the subgraph and not the impact graph, so I'll refactor it to use fetchSubgraph instead

File: src/features/store.ts [hunk 2]
```diff
@@ -1,14 +1,31 @@
...
+				ignoredActions: [
+					'price/setGivPrice',
+					'price/setEthPrice',
+					'price/setXDaiThirdPartTokensPrice',
+					'price/getTokenPrice',
+					'price/getTokenPrice/fulfilled',
+				],
+				ignoredPaths: ['price'],
+			},
+		}),
```
[the remark this answers is not in the document]
Comment (mateodaza) [reply src/features/store.ts:28]: Got it, changed ethPrice and givPrice to always be a string

**Comment by mateodaza** [comment]
@MohammadPCh I added the reducers, not sure why we need the extraReducer as you mention in your comment. 

All other suggestions are solved and ready for review, thank you

**Comment by MohammadPCh** [comment]
> @MohammadPCh I added the reducers, not sure why we need the extraReducer as you mention in your comment.
> 
> All other suggestions are solved and ready for review, thank you

Thank you @mateodaza, You made a Thunk called getTokenPrice. if you want to use it you have to add an extraReducer for it.

**Comment by MohammadPCh** [comment]
#824

**Pull request closed without merging**